skimage/exposure/exposure.py

```diff
@@ -19,7 +19,51 @@
                     'float': dtype_range[np.float64]})
 
 
-def histogram(image, nbins=256):
+def _offset_array(arr, low_boundary, high_boundary):
+    """Offset the array to get the lowest value at 0 if negative.
+    """
+    if low_boundary < 0:
+        offset = low_boundary
+        dyn_range = high_boundary - low_boundary
+        # get smallest dtype that can hold both minimum and offset maximum
+        offset_dtype = np.promote_types(np.min_scalar_type(dyn_range),
+                                        np.min_scalar_type(low_boundary))
+        if arr.dtype != offset_dtype:
+            # prevent overflow errors when offsetting
+            arr = arr.astype(offset_dtype)
+        arr = arr - offset
+    else:
+        offset = 0
+    return arr, offset
+
+
+def _bincount_histogram(image, source_range):
+    """Efficient histogram calculation for a flat image of interfers.
+    """
+    if source_range == 'image':
+        image_min = np.min(image).astype(np.int64)
+        image_max = np.max(image).astype(np.int64)
+        image, offset = _offset_array(image, image_min, image_max)
+        hist = np.bincount(image)
+        bin_centers = np.arange(len(hist)) + offset
+        # clip histogram to start with a non-zero bin
+        idx = np.nonzero(hist)[0][0]
+        hist, bin_centers = hist[idx:], bin_centers[idx:]
+    elif source_range == 'dtype':
```

Comment:
It seems to me like these two paths can be made to share everything except the definition of image_min and image_max.

Reply:
@sciunto I'm not sure that this has been addressed properly?

Reply:
I do not understand what you want to do...

Reply:
Me neither... =P Did you change this code since I wrote the comment? By memory, stuff like `image, offset = ` were also within the `if source_range ==` clauses, so there was a lot of repeated code. But now it's fine! =)